src/firejail/profile.c

```diff
@@ -1706,16 +1706,40 @@ void profile_read(const char *fname) {
 	while (fgets(buf, MAX_READ, fp)) {
 		++lineno;
 		// remove empty space - ptr in allocated memory
-		char *ptr = line_remove_spaces(buf);
-		if (ptr == NULL)
+		char *ptr_clean = line_remove_spaces(buf);
+		if (ptr_clean == NULL)
 			continue;
 
 		// comments
-		if (*ptr == '#' || *ptr == '\0') {
-			free(ptr);
+		if (*ptr_clean == '#' || *ptr_clean == '\0') {
+			free(ptr_clean);
 			continue;
 		}
 
+		// Replace all template keys on line
+		int err;
+		char *ptr;
+
+		fprintf(stderr, "line %s\n", ptr_clean);
+
+		if (strrchr(ptr_clean, '$')) {
+			ptr = template_replace_keys(ptr_clean, &err);
+
+			if (ptr == NULL) {
+				fprintf(stderr, "Ignoring line \"%s\", contains "
+						"invalid template keys (err %d)\n",
```

Comment:
Noting that error code value is not really used by the code itself: as places where those errors are triggered also contain error logging with details -> what are the chances that anybody reading the logs is also interested to know that all that human readable stuff is also known as "(err -127)".

Reply:
Maybe I should in general simplify that error use. Existing code does not use much different error codes but I guess here it could be good to report something that describes the actual cause. I did remove most of that to be as internal in `template.c` for reporting errors, might be enough?